### DISCUSSION TITLE
Pypy getting called

### DISCUSSION THREAD

**Discussion by amirhosseindavoody** [prefix-dev/pixi#1322] [General]
I have a `pixi.toml` file like the following. When I activate the environment using `pixi shell` command and just call python the header shows that I am using PyPy instead of CPython. Can someone tell me why this is happening?

## Pixi.toml file
```sh
[project]
authors = ["LastName, FirstName <my@email.com>"]
channels = ["conda-forge"]
description = "Add a short description here"
name = "src"
platforms = ["linux-64"]
version = "0.1.0"

[system-requirements]
linux = "4.12.14"

[tasks]
format = "ruff format ./"

[dependencies]
jedi = "0.19.1.*"
loguru = "0.7.2.*"
matplotlib = "3.8.1.*"
pandas = "2.1.3.*"
pytest = "7.4.3.*"
python = "3.9.*"
ruff = "0.1.6.*"
scipy = "1.11.3.*"
pyyaml = ">=6.0.1,<6.1"
ipython = ">=8.18.1,<8.19"
```

## Python welcome message:
```sh
Python 3.9.18 | packaged by conda-forge | (c5262994, Oct 26 2023, 08:28:20)
[PyPy 7.3.13 with GCC 12.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>>> 
```

**Comment by ruben-arts** [comment]
That sounds like a weird solution, not sure why that would occur. The example you gave doesn't reproduce it on my machine. Is it reproducible for you?

**Reply by amirhosseindavoody** [reply]
I wouldn't say it is reproducible meaning that I don't know how to reproduce it on any machine. But, it is consistent when I see it.
Let me explain ... our companies compute environment where you request a linux machine and you get one. It is not always the same machine, but all of them have linux suse 12 as their operating system. I don't see this issue on every machine, but when I see it I don't know how to resolve it and I get it everytime I do a `pixi shell`. Even if I remove the `.pixi` folder.

**Reply by ruben-arts** [reply]
Just to make sure some things more clear for me.

Could you share the output of these commands:
- `pixi run which python`
- `which python`
- `pixi run python -VV`
- `python -VV`
- `pixi info`

And could you share the python version from the lockfile (or the whole lockfile), `pixi.lock`.

**Comment by amirhosseindavoody** [comment]
I think I figured why pypy is being installed in the first place. Some of the packages in the lockfile depends on it. I still don't know how they got in the lockfile in the first place. I create a new lockfile from the pixi.toml file and they don't have the pypy dependency.

Anyway, thanks for the help.
